File: iOS/ServiceExtension/NotificationService.m
//
//  NotificationService.m
//  NotificationService
//
//  Created by qgraph on 14/09/16.
//  Copyright © 2016 QuantumGraph. All rights reserved.
//

#import "NotificationService.h"
#import "QGNotificationSdk.h"

@interface NotificationService ()

@end

static NSString *APP_GROUP = @"group.com.company.product.notification";

@implementation NotificationService

- (void)didReceiveNotificationRequest:(UNNotificationRequest *)request withContentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler {
    QGNotificationSdk *qgsdk = [QGNotificationSdk sharedInstanceWithAppGroup:APP_GROUP];
    
    [qgsdk didReceiveNotificationRequest:request withContentHandler:^(UNNotificationContent *content){
        contentHandler(content);
    }];
}


- (void)serviceExtensionTimeWillExpire {
    [[QGNotificationSdk sharedInstanceWithAppGroup:APP_GROUP] serviceExtensionTimeWillExpire];
}

@end

File: iOS/ServiceExtension/NotificationService.swift
//
//  NotificationService.swift
//  TestServiceExtension
//
//  Created by qgraph on 27/09/16.
//  Copyright © 2016 QuantumGraph. All rights reserved.
//

import UserNotifications

class NotificationService: UNNotificationServiceExtension {

    let APP_GROUP = "APP_GROUP"

    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
        
        let qgsdk = QGNotificationSdk.sharedInstance(withAppGroup: APP_GROUP)
        
        qgsdk.didReceive(request, withContentHandler: { content in
            contentHandler(content)
            })
    }
    
    override func serviceExtensionTimeWillExpire() {
        QGNotificationSdk.sharedInstance(withAppGroup: APP_GROUP).serviceExtensionTimeWillExpire()
    }

}
